example/waf/security_application_manager/scopes.go

```diff
@@ -28,52 +37,156 @@ func main() {
 		return
 	}
 
+	fmt.Println("**** CREATE SCOPES ****")
+
+	// Some variables we can take the address of for pointers
+	trueVar := true
+	encodedMessage := base64.StdEncoding.EncodeToString([]byte("hello!"))
+	status404 := 404
+	status200 := 200
+	validForSec := 300
+	redirectURL := "https://www.mysite.com/redirected"
+
+	scope := waf.Scope{
+		Name: "Sample Scope",
+		Host: waf.MatchCondition{
+			Type:              "EM",
+			IsCaseInsensitive: &trueVar,
+			Values:            &[]string{"mysite.com", "mysite2.com"},
+		},
+		Path: waf.MatchCondition{
+			Type:   "EM",
+			Values: &[]string{"/account", "/admin"},
+		},
+		Limits: &[]waf.Limit{
+			{
+				ID: rateRuleID,
+				Action: waf.LimitAction{
+					Name:               "Custom action",
+					DurationSec:        10,
+					ENFType:            "CUSTOM_RESPONSE",
+					ResponseBodyBase64: &encodedMessage,
+					ResponseHeaders:    &map[string]string{"key1": "value1"},
+					Status:             &status404,
+				},
+			},
+		},
+		ACLAuditID: &accessRuleID,
+		ACLProdID:  &accessRuleID,
+		ACLProdAction: &waf.ProdAction{
+			Name:    "Access Rule Action",
+			ENFType: "REDIRECT_302",
+			URL:     &redirectURL,
+		},
+		ProfileAuditID: &managedRuleID,
+		ProfileProdID:  &managedRuleID,
+		ProfileProdAction: &waf.ProdAction{
+			Name:    "Managed Rule Action",
+			ENFType: "BLOCK_REQUEST",
+		},
+		RuleAuditID: &customRuleID,
+		RuleProdID:  &customRuleID,
+		RuleProdAction: &waf.ProdAction{
+			Name:    "Custom Rule Action",
+			ENFType: "ALERT",
+		},
+		BotsProdID: &botRuleID,
+		BotsProdAction: &waf.ProdAction{
+			Name:        "Bot Rule Action",
+			ENFType:     "BROWSER_CHALLENGE",
+			Status:      &status200,
+			ValidForSec: &validForSec,
+		},
+	}
+
+	scopes := waf.Scopes{
+		CustomerID: customerID,
+		Scopes:     []waf.Scope{scope},
+	}
+
+	modifyResp, err := wafService.ModifyAllScopes(scopes)
+
+	if err != nil || !modifyResp.Success {
+		fmt.Printf("Failed to create security application manager configurations (scopes): %+v\n", err)
```

Comment:
If err != nil but modifyResp.Succes is false, do you want to substitute printing err with WAFResponse.Errors or something else?

Reply:
@wdawg33 I believe the err object contains the WAFResponse in JSON format, but i will double check that

Reply:
@stevenpaz gotcha! Wondering then if we could end up in a position where err == nil but modifyResp.Success is false, and if there is anything else we can provide in the returned error message.

Reply:
@wdawg33 I think it might be best to create a new err inside of `ModifyAllScopes` if the API call returns a nil err but WAFResponse.Success == false